## Подготовительные работы.

* Установка необходимых для анализа библиотек.

In [ ]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
display('Все библиотеки импортированы')

* Загрузка датасетов.

In [ ]:
# ВАРИАНТ ПЕРВЫЙ: через функцию, которая будет загружать таблицы по имени
# Функция для загрузки таблицы из CSV
def load_table(table_name):
    print(f"Загружаем таблицу {table_name}...")
    file_path = rf"C:\Users\rit_k\Downloads\bi-analyst-{table_name}.csv"
    df = pd.read_csv(file_path)
    print(f"Таблица {table_name} загружена, размер: {df.shape}")
    return df

# Список ключевых таблиц
tables = ['categories', 'orders', 'order_items', 'products', 'reviews', 'transactions', 'users']


# Загружаем все таблицы в словарь
data = {}
for tbl in tables:
    data[tbl] = load_table(tbl)

Загружаем таблицу categories...
Таблица categories загружена, размер: (1800, 3)
Загружаем таблицу orders...
Таблица orders загружена, размер: (505400, 5)
Загружаем таблицу order_items...
Таблица order_items загружена, размер: (1516751, 5)
Загружаем таблицу products...
Таблица products загружена, размер: (52100, 7)
Загружаем таблицу reviews...
Таблица reviews загружена, размер: (310100, 6)
Загружаем таблицу transactions...
Таблица transactions загружена, размер: (811200, 5)
Загружаем таблицу users...
Таблица users загружена, размер: (111300, 7)


In [ ]:
# ВАРИАНТ ВТОРОЙ: через прямую загрузку
categories = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-categories.csv')
orders = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-orders.csv')
order_items = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-order_items.csv')
products = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-products.csv')
reviews = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-reviews.csv')
transactions = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-transactions.csv')
users = pd.read_csv(r'C:\Users\rit_k\Downloads\bi-analyst-users.csv')
display('Все датасеты загружены')

## Знакомство с датасетами.

### Знакомство с таблицей `categories`.

In [ ]:
# Выводим информацию о датафрейме, первые 5 строк и количество полных дубликатов
display(categories.info())
display(categories.head(5))
display(f'Полных дубликатов: {categories.duplicated().sum()}')

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1800 entries, 0 to 1799
Data columns (total 3 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   category_id         1800 non-null   int64  
 1   name                1798 non-null   object 
 2   parent_category_id  915 non-null    float64
dtypes: float64(1), int64(1), object(1)
memory usage: 42.3+ KB


'Полных дубликатов: 0'

* Таблица `categories` состоит из 3 колонок и 1800 строк;
* Присутствуют пропущенные значения в поле `parent_category_id` - объясняется тем, что соответствующее значение в поле `category_id` уже является родительской;
* ⚠️ Присутствуют пропущенные значения в поле `name`, по условию задачи пока ничего не делаем;
* Отсутствуют полные дубликаты;

### Знакомство с таблицей `orders`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(orders.info())
display(orders.head(5))
display(f'Полных дубликатов: {orders.duplicated().sum()}')

* Таблица `orders` состоит из 5 колонок и 505400 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `order_date` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
orders['order_date'] = pd.to_datetime(orders['order_date'])
display(orders.dtypes)

* ✅ Поле `order_date` преобразовано к datatype типу данных.

In [ ]:
# Проверка отрицательных значений в поле total_amount
display(orders['total_amount'].agg(['min', 'max']))

* Все значения в поле `total_amount` положительные.

### Знакомство с таблицей `order_items`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(order_items.info())
display(order_items.head(5))
display(f'Полных дубликатов: {order_items.duplicated().sum()}')

* Таблица `order_items` состоит из 5 колонок и 1516751 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты.

In [ ]:
# Проверка отрицательных значений в поле price_at_order_time
display(order_items['price_at_order_time'].agg(['min', 'max']))

* Все значения в поле `price_at_order_time` положительные.

### Знакомство с таблицей `products`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(products.info())
display(products.head(5))
display(f'Полных дубликатов: {products.duplicated().sum()}')

* Таблица `products` состоит из 7 колонок и 52099 строк;
* Присутствуют пропущенные значения в поле `title`;
* Отсутствуют полные дубликаты;
* ⚠️ По условию необходимо проверить наличие отрицательной цены и изменить ее на 0.

In [ ]:
# Замена отрицательной цены на 0
products.loc[products['price'] < 0, 'price'] = 0
display(products['price'].agg(['min', 'max']))

* ✅ Все цены положительные.

### Знакомство с таблицей `reviews`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(reviews.info())
display(reviews.head(5))
display(f'Полных дубликатов: {reviews.duplicated().sum()}')

* Таблица `reviews` состоит из 6 колонок и 310000 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `review_date` необходимо преобразовать к datatype типу данных;
* ⚠️ По условию необходимо добавить новый столбец с категоризацией отзывов.

In [ ]:
# Преобразование типа данных
reviews['review_date'] = pd.to_datetime(reviews['review_date'])
display(reviews.dtypes)

* ✅ Поле `review_date` преобразовано к datatype типу данных.

In [ ]:
# Добавление нового столбца с категоризацией рейтинга
reviews['rating_category'] = 'Low'                                 # дефолт
reviews.loc[reviews['rating'] >= 4, 'rating_category'] = 'Medium'
reviews.loc[reviews['rating'] >= 4.5, 'rating_category'] = 'High'
reviews[['rating', 'rating_category']].head(10)

* ✅ Новое поле `rating_category` успешно создано.

### Знакомство с таблицей `transactions`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(transactions.info())
display(transactions.head(5))
display(f'Полных дубликатов: {transactions.duplicated().sum()}')

* Таблица `transactions` состоит из 5 колонок и 811200 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `transaction_type` необходимо преобразовать к category типу данных;
* ⚠️ Поле `transaction_date` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
transactions['transaction_type'] = transactions['transaction_type'].astype('category')
transactions['transaction_date'] = pd.to_datetime(transactions['transaction_date'])
display(transactions.dtypes)

* ✅ Поле `transaction_type` преобразовано к category типу данных;
* ✅ Поле `transaction_date` преобразовано к datatype типу данных.

In [ ]:
# Проверка отрицательных значений в поле total_amount
display(transactions['amount'].agg(['min', 'max']))

* ⚠️ Встречаются отрицательные значения, по условию задачи пока ничего не делаем.

### Знакомство с таблицей `users`.

In [ ]:
# Вывод информации о датафрейме, первые 5 строк и количество полных дубликатов
display(users.info())
display(users.head(5))
display(f'Полных дубликатов: {users.duplicated().sum()}')

* Таблица `users` состоит из 7 колонок и 111300 строк;
* Отсутствуют пропущенные значения;
* Отсутствуют полные дубликаты;
* ⚠️ Поле `user_type` необходимо преобразовать к category типу данных;
* ⚠️ Поле `registration_date` необходимо преобразовать к datatype типу данных.

In [ ]:
# Преобразование типа данных
users['user_type'] = users['user_type'].astype('category')
users['registration_date'] = pd.to_datetime(users['registration_date'])
display(users.dtypes)

* ✅ Поле `user_type` преобразовано к category типу данных;
* ✅ Поле `registration_date` преобразовано к datatype типу данных.